src/openms/include/OpenMS/QC/Suitability.h

```diff
@@ -0,0 +1,173 @@
+    *                FDR        - q-value that should be filtered for
+    *
+    * Result is appended to the result member. This allows for multiple usage.
+    *
+    * Attention: This function will calculate q-values for all peptide hits. Your input
+    *            will not be the same afterwards.
+    *
+    * @param pep_ids      vector containing pepIDs coming from a deNovo+database 
+    *                     identification search without FDR (currently only Comet-support)
+    * @throws             MissingInformation if no target/decoy annotation is found
+    * @throws             MissingInformation if no xcorr is found
+    * @throws             Precondition if a q-value is found in the input
+    */
+    void compute(std::vector<PeptideIdentification>& pep_ids);
+
+    /// return results
+    const std::vector<SuitabilityData>& getResults() const;
+
+  private:
+    /// result vector
+    std::vector<SuitabilityData> results_;
+
+    /**
+    * @brief Calculates the xcorr difference between the top two hits marked as decoy
+    *
+    * Only searches the top ten hits for two decoys. If there aren't two decoys, DBL_MAX
+    * is returned.
+    *
+    * @param pep_id     pepID from where the decoy difference will be calculated
+    * @returns          xcorr difference
+    * @throws           MissingInformation if no target/decoy annotation is found
+    * @throws           MissingInformation if no xcorr is found
+    */
+    double getDecoyDiff_(const PeptideIdentification& pep_id);
+
+    /**
+    * @brief Calculates a xcorr cut-off based on decoy hits
+    *
+    * Decoy differences of all N pepIDs are calculated. The (1-novo_fract)*N highest
+    * one is returned.
+    * It is asssumed that this difference accounts for novo_fract of the re-ranking cases.
+    *
+    * @param pep_ids      vector containing the pepIDs
+    * @param novo_fract   fraction of how many cases, where a de novo peptide scores just higher than the database peptide, will be re-rank
+    * @returns            xcorr cut-off
+    * @throws             IllegalArgument if novo_fract isn't in range [0,1]
+    * @throws             IllegalArgument if novo_fract is to low for a decoy cut-off to be calculated
+    * @throws             MissingInformation if no more than 20 % of the pepIDs have two decoys in there top ten
+    */
+    double getDecoyCutOff_(const std::vector<PeptideIdentification>& pep_ids, double novo_fract);
+
+    /**
+    * @brief Tests if a PeptideHit is considered a deNovo hit
+    *
+    * To test this the function looks into the protein accessions.
+    * If only the deNovo protein is found, 'true' is returned.
+    * If one database protein is found, 'false' is returned.
+    *
+    * @param hit      PepHit in question
+    * @returns        true/false
+    */
+    bool isNovoHit_(const PeptideHit& hit);
+
+    /**
+    * @brief Tests if a PeptideHit has a higher q-value the given FDR
```

Comment:
`has a higher q-value than a given FDR, i.e. does not pass FDR.`

Reply:
... threshold